cmd/kubelet/app/server.go

```diff
@@ -292,9 +291,13 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 			// run the kubelet
 			if err := Run(ctx, kubeletServer, kubeletDeps, utilfeature.DefaultFeatureGate); err != nil {
 				klog.ErrorS(err, "Failed to run kubelet")
-				os.Exit(1)
+				return err
 			}
+
+			return nil
 		},
+		SilenceErrors: true,
+		SilenceUsage:  true,
```

Comment:
add `SilenceUsage=true` or usage would be printed twice. See https://github.com/kubernetes/kubernetes/pull/104774/files#diff-3a14a9838132c6668f4d100c26afa05b471b149c469edfece9b84fcd058c8a4bR159

Reply:
/lgtm
Nice catch